### DISCUSSION TITLE
How can we restart explorer.exe so that ExplorerPatcher does not trigger "explorer crashed"?

### DISCUSSION THREAD

**Discussion by schittli** [valinet/ExplorerPatcher#3395] [General]
Good evening

**Thanks a lot for sharing this great work!, it's a shame that Microsoft does not have the knowhow to do its work.**

I have a small question:
Explorer has had (for years) very embarrassing problems, so it can happen that Windows suddenly has a higher base CPU load. If you then kill explorer.exe and restart it, everything runs fine again.
Therefore, I sometimes start this PowerShell command to restart explorer:

`taskkill /f /im explorer.exe; Start-Sleep -Milliseconds 1500; explorer.exe`

Unfortunately, this triggers the ExplorerPatcher popup dialog "I'm sorry, explorer crashed".

Is there a way to disable the explorer crash detector / this popup dialog?

Thanks a lot, kind regards,
Thomas

**Comment by pyrates999** [comment]
No, because EP needs to keep track of how many times explorer.exe has crashed and not load EP if it continually happens.

If your computer has high cpu usage and only killing explorer.exe fixes it, I would do a chkdsk against all your drives.  There might be file system errors.

**Reply by schittli** [reply]
Thank you very much, @pyrates999 for your answer!

> No (it is not possible to disable the EP explorer crashed trigger), 
> because EP needs to track how often explorer.exe has crashed and not load EP if it continually happens.

I take your answer as yes, it would be possible 😃 
* Because if the point is that EP knows that the Explorer has crashed regularly, then EP must record how many crashes there have been and over what period.
* Therefore, it would be possible to enable EP to behave in such a way that it does not display the Explorer crashed dialog in the event of one Explorer crash 😃 


> If your computer has high CPU usage and only killing explorer.exe fixes it, I would do a chkdsk against all your drives. There might be file system errors.

This is an excellent tip, thank you very much!

However, at work, many employees have this problem (we have 750 employees), but to be on the safe side I checked all my disks and let chkdsk scan for bad blocks on my computer. 

The result:
* There was not a single problem with the disks
* This means: The Microsoft Windows Explorer actually has a fundamental problem where it suddenly slows down the system very noticeably and after a kill of explorer.exe and a restart, the problem is solved immediately.

Therefore, our tip when employees report Windows performance problems is to kill the Explorer.exe and restart it. Our experience is that people often think: "That's magic!" 😄 


Thanks a lot, kind regards,
Thomas

**Comment by Amrsatrio** [comment]
You can set HKCU\Software\ExplorerPatcher\CrashCounterDisabled to DWORD with value 1. Though however note that the 10 start menu animations will not be made proper as a safety measure.

**Reply by schittli** [reply]
Thanks a lot for your tip!, @Amrsatrio 
that's great to hear, I'll test it right away.

Luckily I always disable animations because I don't want to watch and wait for them to finish 😄 

Thanks a lot, kind regards,
Thomas

**Comment by Chesterboateng** [comment]
does it work?
